keras_hub/src/models/qwen3/qwen3_backbone.py

```diff
@@ -0,0 +1,332 @@
+            such as softmax and layer normalization, will always be done at
+            float32 precision regardless of dtype.
+        tie_word_embeddings (bool, optional): Whether to tie input and output
+            embeddings. Defaults to `True`.
+        use_sliding_window_attention (bool, optional): Whether to use sliding
+            window attention for efficient processing of long sequences.
+            Defaults to `False`.
+        sliding_window_size (int, optional): Size of the sliding window for
+            attention when enabled. Defaults to `32768`.
+
+    Examples:
+
+    ```python
+    input_data = {
+        "token_ids": np.ones(shape=(1, 12), dtype="int32"),
+        "padding_mask": np.array([[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0]]),
+    }
+
+    # Pretrained Qwen3 decoder.
+    model = keras_hub.models.Qwen3Backbone.from_preset("qwen32.5_0.5b_en")
+    model(input_data)
+
+    # Randomly initialized Qwen3 decoder with custom config.
+    model = keras_hub.models.Qwen3Backbone(
+        vocabulary_size=10,
+        hidden_dim=512,
+        num_layers=2,
+        num_query_heads=32,
+        num_key_value_heads=8,
+        intermediate_dim=1024,
+        layer_norm_epsilon=1e-6,
+        dtype="float32"
+    )
+    model(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        vocabulary_size,
+        num_layers,
+        num_query_heads,
+        num_key_value_heads,
+        head_dim,
+        hidden_dim,
+        intermediate_dim,
+        rope_max_wavelength=10000,
+        rope_scaling_factor=1.0,
+        layer_norm_epsilon=1e-6,
+        dropout=0,
+        dtype=None,
+        tie_word_embeddings=True,
+        use_sliding_window_attention=False,
+        sliding_window_size=32768,
+        max_window_layers=28,
+        **kwargs,
+    ):
+        # === Layers ===
+        self.token_embedding = ReversibleEmbedding(
+            input_dim=vocabulary_size,
+            output_dim=hidden_dim,
+            tie_weights=tie_word_embeddings,
+            embeddings_initializer=_qwen3_kernel_initializer(stddev=0.01),
+            dtype=dtype,
+            name="token_embedding",
+        )
+        self.transformer_layers = []
+        for i in range(num_layers):
+            layer = Qwen3TransformerDecoder(
+                intermediate_dim=intermediate_dim,
+                head_dim=head_dim,
+                num_query_heads=num_query_heads,
+                num_key_value_heads=num_key_value_heads,
+                rope_max_wavelength=rope_max_wavelength,
+                rope_scaling_factor=rope_scaling_factor,
+                layer_norm_epsilon=layer_norm_epsilon,
+                activation=ops.silu,
+                kernel_initializer=_qwen3_kernel_initializer(stddev=0.02),
+                dropout=dropout,
+                dtype=dtype,
+                use_sliding_window_attention=use_sliding_window_attention,
+                sliding_window_size=sliding_window_size,
+                layer_index=i,
```

Comment:
Same here, remove layer_index, max_window_layers, use_sliding_window_attention from the decoder layer.

Fine to keep `max_window_layers` for the backbone, but maybe `num_sliding_window_layers` is more accurate? It's not the max it's the actual count right?